### ISSUE TITLE
"cannot import name cast_unicode_py2" from ptshell.py

### ISSUE THREAD

**Issue by stegrams** [jupyter/jupyter_console#198]
Because of jupyter_console/ptshell.py line 22, I was unable to start the coconut --ipython console.
In order to succeed, I changed the file as follows:
```python
from IPython.utils.py3compat import input
from ipython_genutils.py3compat import cast_unicode_py2
```

**Comment by dylanfprice** [comment]
I am also seeing this error when I try to install and run jupyter-console:

```bash
$ python3 -m venv test_venv
$ source test_venv/bin/activate
$ pip install jupyter-console
Collecting jupyter-console
  Using cached https://files.pythonhosted.org/packages/cb/ee/6374ae8c21b7d0847f9c3722dcdfac986b8e54fa9ad9ea66e1eb6320d2b8/jupyter_console-6.0.0-py2.py3-none-any.whl
Collecting jupyter-client (from jupyter-console)
  Using cached https://files.pythonhosted.org/packages/13/81/fe0eee1bcf949851a120254b1f530ae1e01bdde2d3ab9710c6ff81525061/jupyter_client-5.3.4-py2.py3-none-any.whl
Collecting ipykernel (from jupyter-console)
  Using cached https://files.pythonhosted.org/packages/e1/92/8fec943b5b81078399f969f00557804d884c96fcd0bc296e81a2ed4fd270/ipykernel-5.1.3-py3-none-any.whl
Collecting pygments (from jupyter-console)
  Using cached https://files.pythonhosted.org/packages/be/39/32da3184734730c0e4d3fa3b2b5872104668ad6dc1b5a73d8e477e5fe967/Pygments-2.5.2-py2.py3-none-any.whl
Collecting ipython (from jupyter-console)
  Using cached https://files.pythonhosted.org/packages/05/d7/77b7a1988c99227f52402f93fb0f7e88c97239960516f53907ebbc44149c/ipython-7.11.0-py3-none-any.whl
Collecting prompt-toolkit<2.1.0,>=2.0.0 (from jupyter-console)
  Using cached https://files.pythonhosted.org/packages/87/61/2dfea88583d5454e3a64f9308a686071d58d59a55db638268a6413e1eb6d/prompt_toolkit-2.0.10-py3-none-any.whl
Collecting python-dateutil>=2.1 (from jupyter-client->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/d4/70/d60450c3dd48ef87586924207ae8907090de0b306af2bce5d134d78615cb/python_dateutil-2.8.1-py2.py3-none-any.whl
Collecting jupyter-core>=4.6.0 (from jupyter-client->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/fb/82/86437f661875e30682e99d04c13ba6c216f86f5f6ca6ef212d3ee8b6ca11/jupyter_core-4.6.1-py2.py3-none-any.whl
Collecting tornado>=4.1 (from jupyter-client->jupyter-console)
Collecting pyzmq>=13 (from jupyter-client->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/bf/dd/547bee2080beefe6546c332422cb0f189b3cd71cfff048146cd9ac3751cc/pyzmq-18.1.1-cp37-cp37m-manylinux1_x86_64.whl
Collecting traitlets (from jupyter-client->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/ca/ab/872a23e29cec3cf2594af7e857f18b687ad21039c1f9b922fac5b9b142d5/traitlets-4.3.3-py2.py3-none-any.whl
Collecting pickleshare (from ipython->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/9a/41/220f49aaea88bc6fa6cba8d05ecf24676326156c23b991e80b3f2fc24c77/pickleshare-0.7.5-py2.py3-none-any.whl
Collecting jedi>=0.10 (from ipython->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/e9/97/55e575a5b49e5c3df9eb3c116c61021d7badf556c816be13bbd7baf55234/jedi-0.15.2-py2.py3-none-any.whl
Collecting decorator (from ipython->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/8f/b7/f329cfdc75f3d28d12c65980e4469e2fa373f1953f5df6e370e84ea2e875/decorator-4.4.1-py2.py3-none-any.whl
Collecting backcall (from ipython->jupyter-console)
Collecting pexpect; sys_platform != "win32" (from ipython->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/0e/3e/377007e3f36ec42f1b84ec322ee12141a9e10d808312e5738f52f80a232c/pexpect-4.7.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools>=18.5 in ./test_venv/lib/python3.7/site-packages (from ipython->jupyter-console) (40.8.0)
Collecting wcwidth (from prompt-toolkit<2.1.0,>=2.0.0->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/7e/9f/526a6947247599b084ee5232e4f9190a38f398d7300d866af3ab571a5bfe/wcwidth-0.1.7-py2.py3-none-any.whl
Collecting six>=1.9.0 (from prompt-toolkit<2.1.0,>=2.0.0->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting ipython-genutils (from traitlets->jupyter-client->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/fa/bc/9bd3b5c2b4774d5f33b2d544f1460be9df7df2fe42f352135381c347c69a/ipython_genutils-0.2.0-py2.py3-none-any.whl
Collecting parso>=0.5.2 (from jedi>=0.10->ipython->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/9b/b0/90353a5ece0987279837835224dead0c424833a224195683e188d384e06b/parso-0.5.2-py2.py3-none-any.whl
Collecting ptyprocess>=0.5 (from pexpect; sys_platform != "win32"->ipython->jupyter-console)
  Using cached https://files.pythonhosted.org/packages/d1/29/605c2cc68a9992d18dada28206eeada56ea4bd07a239669da41674648b6f/ptyprocess-0.6.0-py2.py3-none-any.whl
Installing collected packages: six, python-dateutil, ipython-genutils, decorator, traitlets, jupyter-core, tornado, pyzmq, jupyter-client, pickleshare, parso, jedi, pygments, backcall, wcwidth, prompt-toolkit, ptyprocess, pexpect, ipython, ipykernel, jupyter-console
Successfully installed backcall-0.1.0 decorator-4.4.1 ipykernel-5.1.3 ipython-7.11.0 ipython-genutils-0.2.0 jedi-0.15.2 jupyter-client-5.3.4 jupyter-console-6.0.0 jupyter-core-4.6.1 parso-0.5.2 pexpect-4.7.0 pickleshare-0.7.5 prompt-toolkit-2.0.10 ptyprocess-0.6.0 pygments-2.5.2 python-dateutil-2.8.1 pyzmq-18.1.1 six-1.13.0 tornado-6.0.3 traitlets-4.3.3 wcwidth-0.1.7
(test_venv) $ jupyter console
Traceback (most recent call last):
  File "/test_venv/bin/jupyter-console", line 6, in <module>
    from jupyter_console.app import main
  File "/test_venv/lib/python3.7/site-packages/jupyter_console/app.py", line 26, in <module>
    from jupyter_console.ptshell import ZMQTerminalInteractiveShell
  File "/test_venv/lib/python3.7/site-packages/jupyter_console/ptshell.py", line 22, in <module>
    from IPython.utils.py3compat import cast_unicode_py2, input
ImportError: cannot import name 'cast_unicode_py2' from 'IPython.utils.py3compat' (/test_venv/lib/python3.7/site-packages/IPython/utils/py3compat.py)
```

**Comment by moritzschaefer** [comment]
This error is super weird in my eyes:

The import has "always" been named "cast_unicode_py2" (for 4 years now: https://github.com/jupyter/jupyter_console/blame/5.x/jupyter_console/ptshell.py).
However, in ipython.utils.py3compat, the "_py2 has never been ther (https://github.com/ipython/ipython/blame/master/IPython/utils/py3compat.py for 9 years). 
Could someone clarify please?

**Comment by stegrams** [comment]
@moritzschaefer the ```cast_unicode_py2``` was there until 29 days ago when removed from [this PR](https://github.com/ipython/ipython/commit/a089b78315ef964f8552ffa52db569a8a924bfc2).
Before that, ```cast_unicode_py2``` was an identity function that was just returned its parameter.

**Comment by alaskamike** [comment]
I just changed cast_unicode_py2 to cast_unicode and now it runs fine.

**Comment by rgbkrk** [comment]
I've now shipped this as `6.1.0`
